packages/headless/src/features/search/search-request.ts

```diff
@@ -70,6 +70,8 @@ export const buildSearchRequest = async (
       facetOptions: {freezeFacetOrder: state.facetOptions.freezeFacetOrder},
     }),
     ...(state.folding?.enabled && {
+      // TODO KIT-3082 remove the comment below; it's only to demonstrate how the spread syntax can cause issues
```

Comment:
@louis-bompart if you uncomment this line and comment the next one, you'll see there are no compilation errors, and unit tests will all pass.